File: src/main/webapp/public/js/enums/orderStatus.js
const OrderStatus = Object.freeze({
    WAITING_PAYMENT: { name: 'Aguardando Pagamento', code: 1 },
    PAID: { name: 'Pago', code: 2 },
    SHIPPED: { name: 'Enviado', code: 3 },
    DELIVERED: { name: 'Entregue', code: 4 },
    CANCELED: { name: 'Cancelado', code: 5 }
});

File: src/main/java/com/servlet_ordering_system/models/vos/enums/OrderStatus.java
package com.servlet_ordering_system.models.vos.enums;


public enum OrderStatus {
    WAITING_PAYMENT(1),
    PAID(2),
    SHIPPED(3),
    DELIVERED(4),
    CANCELED(5);

    private final int code;

    private OrderStatus(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public static OrderStatus valueOf(int code) {
        for(OrderStatus status : OrderStatus.values()) {
            if (status.getCode() == code) {
                return status;
            }
        }

        return  null;
    }
}
